### ISSUE TITLE
Is it possible to use my own dataset for image classification?

### ISSUE THREAD

**Issue by test16553** [facebookresearch/dinov2#199]
Hello, is it possible to use a custom dataset for image classification? Approximate process: call the extract_features function of dinov2 to extract the feature vector of the target picture, and then call the knn algorithm of dinov2 to classify based on the obtained feature vector of the target picture and its own data set, and determine the label of the target picture. If this pipeline doesn't work, do you have an example of image classification using a new dataset？

**Comment by patricklabatut** [comment]
Yes, you should even be able to re-use the provided k-NN evaluation, see the [[code]](https://github.com/facebookresearch/dinov2/blob/main/dinov2/eval/knn.py) and the associated [[instructions]](https://github.com/facebookresearch/dinov2#k-nn-classification-on-imagenet-1k). You would simply need to provide your dataset to replace ImageNet-1k.

**Comment by shersoni610** [comment]
Hello,

I am running the code on a Mac mini with 16gb ram. Does any one have an idea on how many images and classes we can use to classify images? Is there an out of core implementation of the “fit” method.
